src/policy/semantic.rs

```diff
@@ -188,7 +188,7 @@ impl<Pk: MiniscriptKey> Policy<Pk> {
     // Witness is currently encoded as policy. Only accepts leaf fragment and
     // a normalized policy
     fn satisfy_constraint(self, witness: &Policy<Pk>, available: bool) -> Policy<Pk> {
-        debug_assert!(self.clone().normalized() == self.clone());
+        debug_assert!(self.clone().normalized() == self);
```

Comment:
Interesting, so `self` is not moved here?

Because we are using self couple of lines below.